src/providers/piwik/angulartics2-piwik.ts

```diff
@@ -60,14 +60,37 @@ export class Angulartics2Piwik {
     }
   }
 
+  /**
+   * Sets custom dimensions if at least one property has the key "dimension<n>",
+   * e.g. dimension10. If there are custom dimensions, any other property is ignored.
+   *
+   * If there are no custom dimensions in the given properties object, the properties
+   * object is saved as a custom variable.
+   *
+   * If in doubt, prefer custom dimensions.
+   * @see https://piwik.org/docs/custom-variables/
+   */
   setUserProperties(properties: any) {
     try {
-      _paq.push(['setCustomVariable', properties]);
+      const dimensions = this.setCustomDimensions(properties);
```

Comment:
If dimensions exist, set them respectively. If there are no dimensions, then set a custom variable (old behavior). This is equivalent to the google analytics provider.